datafusion/sql/src/unparser/dialect.rs

```diff
@@ -218,6 +218,45 @@ pub trait Dialect: Send + Sync {
     fn timestamp_with_tz_to_string(&self, dt: DateTime<Tz>, _unit: TimeUnit) -> String {
         dt.to_string()
     }
+
+    /// Whether the dialect supports an empty select list such as `SELECT FROM table`.
+    ///
+    /// An empty select list returns rows without any column data, which is useful for:
+    /// - Counting rows: `SELECT FROM users WHERE active = true` (combined with `COUNT(*)`)
+    /// - Testing row existence without retrieving column data
+    /// - Performance optimization when only row counts or existence checks are needed
+    ///
+    /// # Supported Databases
+    ///
+    /// - **PostgreSQL**: Fully supported. Returns rows with zero columns.
+    ///
+    /// # Unsupported Databases
+    ///
+    /// Most other SQL databases require at least one column in the SELECT list:
+    /// - MySQL
+    /// - SQLite
+    /// - SQL Server
+    /// - Oracle
+    /// - DuckDB
+    ///
+    /// For these databases, the unparser falls back to `SELECT 1 FROM table`.
+    ///
+    /// # Default
+    ///
+    /// Returns `false` for maximum compatibility across SQL dialects.
```

Comment:
I don't know if we should include this detail about specific databases here; ideally their own dialects would implement this to show their support. We can just leave it at what the default behaviour is.